cellprofiler/modules/savecroppedobjects.py

```diff
@@ -126,67 +210,64 @@ def run(self, workspace):
+                    nested_folder, "{}{}_{}".format(input_filename + "_" if self.use_filename else "", self.objects_name.value, label)
                 )
 
-                skimage.io.imsave(
-                    filename, skimage.img_as_ubyte(mask), check_contrast=False
+            elif not self.nested_save.value:
+                filename = os.path.join(
+                    directory, "{}{}_{}".format(input_filename + "_" if self.use_filename else "", self.objects_name.value, label)
                 )
 
-            elif self.file_format.value == O_TIFF_8:
-                filename = os.path.join(
-                    directory, "{}_{}.{}".format(self.objects_name.value, label, "tiff")
+            if self.file_format.value == O_PNG:
+                save_filename = filename + ".{}".format(O_PNG)
+
+                skimage.io.imsave(
+                    save_filename, 
+                    skimage.img_as_ubyte(mask), 
+                    check_contrast=False
                 )
 
+            elif self.file_format.value == O_TIFF_8:
+                save_filename = filename +".{}".format("tiff")
+                
                 skimage.io.imsave(
-                    filename,
+                    save_filename,
                     skimage.img_as_ubyte(mask),
                     compress=6,
                     check_contrast=False,
                 )
 
             elif self.file_format.value == O_TIFF_16:
-                filename = os.path.join(
-                    directory, "{}_{}.{}".format(self.objects_name.value, label, "tiff")
-                )
-
+                save_filename = filename + ".{}".format("tiff")
+                
                 skimage.io.imsave(
-                    filename,
+                    save_filename,
                     skimage.img_as_uint(mask),
                     compress=6,
                     check_contrast=False,
                 )
 
-            filenames.append(filename)
+            filenames.append(save_filename)
 
         if self.show_window:
             workspace.display_data.filenames = filenames
 
-    def settings(self):
-        settings = [
-            self.objects_name,
-            self.directory,
-            self.file_format,
-            self.export_option,
-            self.image_name,
-        ]
-
-        return settings
-
-    def visible_settings(self):
-        result = [
-            self.export_option,
-            self.objects_name,
-            self.directory,
-            self.file_format,
-        ]
-
-        if self.export_option.value == SAVE_PER_OBJECT:
-            result += [self.image_name]
-
-        return result
+    def upgrade_settings(self, setting_values, variable_revision_number, module_name):
+        if variable_revision_number < 3:
+            # Older module version, revert to not using file names in output crops
+            # Also, reorder setting_values
```

Comment:
Absolutely fine to reorder, but hard to follow the changes, can you add one more line of comment saying what you're reordering from and what you're reordering to? I can see it in the diff right here right now, of course, but for future folks reading the code but not looking at the specific PR that introduced the change

Reply:
Thanks!